### PR TITLE
🩹 BUG: Fix Parsing Missing Omero Version NGFF Metadata

### PR DESCRIPTION
**Pull request by John-P**
# To-Do

- [x] Handle missing version keys and null values.
- [x] Add test for missing value
- [x] Add test for non-numeric values e.g. "0.5-dev" [see current spec doc](https://ngff.openmicroscopy.org/latest/#omero-md)
- [x] Warn if version < or > 0.4
- [x] Warn for inconsistent versions

### CODE REVIEW

File: tiatoolbox/wsicore/wsireader.py [hunk 1]
```diff
@@ -136,11 +136,14 @@ def is_ngff(path: pathlib.Path, min_version: Tuple[int, ...] = (0, 4)) -> bool:
         tuple(int(part) for part in scale.get("version", "").split("."))
         for scale in multiscales
     )
-    omero_version = tuple(int(part) for part in omero.get("version", "").split("."))
+    omero_version = omero.get("verion")
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:139]:
```suggestion
    omero_version = omero.get("version")
```

File: tiatoolbox/wsicore/wsireader.py [hunk 2]
```diff
@@ -136,11 +136,14 @@ def is_ngff(path: pathlib.Path, min_version: Tuple[int, ...] = (0, 4)) -> bool:
...
+    if omero_version:
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:140]: shouldn't it log a warning if this is empty?

Comment (John-P) [reply tiatoolbox/wsicore/wsireader.py:140]: It's perfectly valid to be empty according to the spec

File: tiatoolbox/wsicore/wsireader.py [hunk 3]
```diff
@@ -36,6 +37,8 @@
 Bounds = Tuple[Number, Number, Number, Number]
 IntBounds = Tuple[int, int, int, int]
 Resolution = Union[Number, Tuple[Number, Number], np.ndarray]
+MIN_NGFF_VERSION = Version("0.4")
+MAX_NGFF_VERSION = Version("0.4")
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:41]: is this not quite strict?

Comment (John-P) [reply tiatoolbox/wsicore/wsireader.py:41]: Yes, intentionally. It's not been tested with anything other than 0.4 so we cannot say we suppport it

Comment (John-P) [reply tiatoolbox/wsicore/wsireader.py:41]: Besides, support for new versions is quite far beyond the scope of this PR.

**Pull request merged**